src/components/BlogPostCard.vue

```diff
@@ -0,0 +1,157 @@
+<template>
+  <div class="card-container">
+    <img :src="user.image" alt="Post User's Profile Picture" class="avatar-img" />
+    <div class="post-content-container">
+      <div class="user-header-container">
+        <p class="user-name-text">{{ user.firstName }} {{ user.lastName }}</p>
+        <p class="user-email-text">&centerdot; {{ user.email }}</p>
+      </div>
+      <p class="post-content-text">{{ props.post.body }}</p>
+      <p class="tags-text" v-if="props.post?.tags?.length">
+        {{ allTags.join(' &centerdot; ') }}
+      </p>
+      <div class="actions-container">
+        <div
+          class="default-action-btn"
+          @click="toggleReaction"
+          :class="{ 'has-reacted': hasReacted }"
+        >
+          <FontAwesomeIcon :icon="[hasReacted ? 'fas' : 'far', 'heart']" />
+          <a onclick="">Reacts {{ post.reactions ? `(${post.reactions})` : '' }}</a>
+        </div>
+        <div class="default-action-btn">
+          <FontAwesomeIcon :icon="['far', 'comment-dots']" />
+          <a onclick="">Comment</a>
+        </div>
+        <div v-if="props.post?.userId === props.currentUserId" class="delete-btn">
+          <FontAwesomeIcon :icon="['far', 'trash-can']" />
+          <a v-on:click="deletePost">Delete</a>
+        </div>
+      </div>
+    </div>
+    <div class="date-container">{{ postDateStr }}</div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+import type { PropType } from 'vue'
+import type { BlogPost, BlogUser } from '@/store/types'
+import { useStore } from 'vuex'
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+
+const props = defineProps({
+  post: {
+    type: Object as PropType<BlogPost>,
+    required: true
+  },
+  currentUserId: {
+    type: String,
+    required: true
+  }
+})
+const store = useStore()
+const postDateStr = props.post?.createDate
+  ? new Date(props.post?.createDate).toLocaleDateString()
+  : ''
+const user = computed<BlogUser>(() => store.getters.usersById[props.post?.userId])
+const hasReacted = computed(
+  () => store.getters.reactionsByUser[props.currentUserId]?.[props.post?.postId]
+)
+const allTags = [
+  user.value.addressCity,
+  user.value.companyName,
+  ...(props.post?.tags ?? [])
+].filter((tag) => tag?.length)
```

Comment:
Should this be in a `ref`? Will the `allTags` have the same value regardless if `props.post?.tags` is updated?

Reply:
if it doesn't need to see changes after mount, that's okay

var that uses a computedValue == another computedValue
-> might need a structure change

 function that returns the desired var based on computedvar